docs/howto/system_testing.md

```diff
@@ -599,6 +599,80 @@ A sample workflow would look like:
+1. Setup:
+    - Start the service to be tested with a given configuration (`data_stream/<name>/_dev/test/sytem/test-name-config.yml`) and variant (`_dev/deploy/variants.yml`).
+    - Build and install the package
+    - Create the required resources in Elasticsearch to configure the ingestion of data through the Elastic Agent.
+        - test policy
+        - add package data stream to the test policy
+    - Ensure agent is enrolled.
+    - Ensure that the data stream has no old documents.
+    - Assign policy to the agent.
+    - Ensure there are hits received in Elasticsearch in the package data stream.
+2. Run the tests (validate fields, check transforms, etc.)
+    - Validate fields (e.g. mappings)
+    - Assert number of hit counts.
+    - Check transforms.
+3. Tear Down:
+    - Rollback all the changes in Elasticsearch:
+        - Restore previous policy to the agent.
+        - Delete the test policy.
+        - Uninstall the package.
+    - Stop the service (e.g. container) to be tested.
+    - Wipe all the documents of the data stream.
+
+This process is repeated for each combination of:
+- data stream `D`, if the package is of `integration` type.
+- configuration file defined under `_dev/test/sytem` folder.
+- variant (`_dev/deploy/variants.yml`).
+
+It's possible also to run these steps independently. For that it is required to set which configuration file (`--config-file`)
+and which variant (`--variant`), if any, is going to be used to start and configure the service for these tests.
+
+Then, each step can be run using one of these flags:
+- Run the setup (`--setup`), after this command is executed:
+    - Service container will be kept running.
+    - Elastic Agent is going to be sending documents to Elasticsearch.
+- Run just the tests (`--no-provision`), after this command is executed:
+    - Service container will not be stopped.
+    - Documents in the Data stream will be deleted, so the tests will use the latest documents sent.
+    - Elastic Agent is going to be sending documents to Elasticsearch.
+    - NOTE: This command can be run several times.
+- Run just the setup (`--tear-down`), after this command:
+    - Service container is going to be stopped.
+    - All changes in Elasticsearch are rollback.
+    - Package Data stream is deleted.
+
+
+Examples:
+
+```shell
+# Start Elastic stack as usual
+elastic-package stack up -v -d
+
+# Testing a package using a configuration file and a variant (e.g. mysql)
+elastic-package test system -v --config-file $(pwd)/data_stream/status/_dev/test/system/test-default-config.yml --variant percona_8_0_36 --setup
+elastic-package test system -v --config-file $(pwd)/data_stream/status/_dev/test/system/test-default-config.yml --variant percona_8_0_36 --no-provision
+elastic-package test system -v --config-file $(pwd)/data_stream/status/_dev/test/system/test-default-config.yml --variant percona_8_0_36 --tear-down
```

Comment:
Also for the tear down, could we make it to don't require config file or variant and just tear down whatever was started?

Reply:
Currently, all these operations are tied , config file and variants are required to create the service options and test config.
```go
	serviceOptions, ctxt, err := r.createConfigService(r.variants[0])
	if err != nil {
		return result.WithError(err)
	}

	testConfig, err := newConfig(r.options.ConfigFilePath, ctxt, r.variants[0])
	if err != nil {
		return nil, fmt.Errorf("unable to load system test case file '%s': %w", r.options.ConfigFilePath, err)
	}
	logger.Debugf("Using config: %q", testConfig.Name())
```

And those are used to prepare the scenario to run the requested operation
```go
	scenario, err := r.prepareScenario(testConfig, ctxt, serviceOptions)
```

Not sure how easy or difficult it would be to do that. Maybe I can write also to a file the `serviceOptions` struct and read it from there. This variable is built here:
https://github.com/elastic/elastic-package/blob/c3c9565314a815f81a3ef9dacb4d6f1c0cdc8df5/internal/testrunner/runners/system/runner.go#L253

Reply:
Latest changesets pushed allowed to remove config-file flag for `--no-provision` and `--tear-down` flags.

It required some changes:
- Initialize `locationsManager` in the `init()` function.
- Write the `testConfig` variable to file at the end of the setup process.
    - This allows us to get the variant and config file in `--no-provision` and `--tear-down` modes
- Read one of the files written in the setup dir in `cmd/testrunner.go`, since it is required to know the configuration file path to get the data stream (and test folders).
    - Add a new method to the interface to allow this.